webapp/channels/src/components/form_error.tsx

```diff
@@ -11,71 +11,70 @@ type Props = {
-
-        if (!error && errors.length === 0) {
-            return null;
-        }
+const FormError: React.FC<Props> = ({
+    error = null,
+    errors = [],
+    iconClassName,
+    margin,
+    textClassName,
+    type,
+}) => {
+    if (!error && errors.length === 0) {
+        return null;
+    }
 
-        // look for the first truthy error to display
-        let message = error;
+    // look for the first truthy error to display
+    let message = error;
 
-        if (!message) {
-            for (const err of errors) {
-                if (err) {
-                    message = err;
-                }
+    if (!message) {
+        for (const err of errors) {
+            if (err) {
+                message = err;
+                break;
             }
         }
+    }
 
-        if (!message) {
-            return null;
-        }
-
-        if (type === 'modal') {
-            return (
-                <div className='form-group'>
-                    <label className='col-sm-12 has-error'>
-                        {message}
-                    </label>
-                </div>
-            );
-        }
+    if (!message) {
+        return null;
+    }
 
-        if (type === 'backstage') {
-            return (
-                <div className='pull-left has-error'>
-                    <label className='control-label'>
-                        {message}
-                    </label>
-                </div>
-            );
-        }
+    if (type === 'modal') {
+        return (
+            <div className='form-group'>
+                <label className='col-sm-12 has-error'>{message}</label>
+            </div>
+        );
+    }
 
-        if (margin) {
-            return (
-                <div className='form-group has-error'>
-                    <label className='control-label'>
-                        {message}
-                    </label>
-                </div>
-            );
-        }
+    if (type === 'backstage') {
+        return (
+            <div className='pull-left has-error'>
+                <label className='control-label'>{message}</label>
+            </div>
+        );
+    }
 
+    if (margin) {
         return (
-            <div className={`col-sm-12 ${textClassName || 'has-error'}`}>
-                <label className='control-label'>
-                    <i className={`fa ${iconClassName || 'fa-exclamation-circle'}`}/> {message}
-                </label>
+            <div className='form-group has-error'>
+                <label className='control-label'>{message}</label>
             </div>
         );
     }
-}
+
+    return (
+        <div className={`col-sm-12 ${textClassName || 'has-error'}`}>
+            <label className='control-label'>
+                <i
+                    className={`fa ${iconClassName || 'fa-exclamation-circle'}`}
+                />{' '}
+                {message}
+            </label>
+        </div>
+    );
+};
+
+export default FormError;
```

Comment:
It used to be a PureComponent, so let's use React.memo here.
```suggestion
export default React.memo(FormError);
```